controller/registries/consul/config.go

```diff
@@ -129,14 +153,18 @@ func (reg *Consul) Start(c registrytype.RegistryConfig) error {
 	go func() {
 		reg.logger.Infof("start refreshing services")
 		q := &consulapi.QueryOptions{
-			WaitTime: dur,
+			WaitTime:   dur,
+			Namespace:  config.Namespace,
+			Datacenter: config.DataCenter,
+			Token:      config.Token,
 		}
 		for {
+
 			select {
 			case <-reg.done:
 				reg.logger.Infof("stop refreshing services")
-				return
-
+				//wait to retry
+				time.Sleep(SleepTime)
```

Comment:
Would be better to use the configured duration as the sleep time, so it could be adaptable. For example, sleep N seconds when the WaitTime is N.